### PR TITLE
Uda prep

### PR DESCRIPTION
**Pull request by joakim-hove**
Have just extracted a bunch of quite trivial commits from: #774

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
I have too many moving parts now - will self merge this and #813 quite soon.

File: src/opm/parser/eclipse/Deck/DeckItem.cpp [hunk 1]
```diff
@@ -294,6 +294,16 @@ void DeckItem::push_backDimension( const Dimension& active,
         const bool dim_inactive = du.empty()
             || this->defaultApplied( du.size() - 1 );
 
+        // The data model when it comes to UDA values, dimensions for vectors
+        // and so on is stretched beyond the breaking point. It is a *really*
+        // hard assumption here that UDA values only apply to scalar values.
+        if (du.size() > 1)
+            throw std::logic_error("Internal program meltdown - we do not handle non-scalar UDA values");
+
+        // This is just so broken ....
```
Comment (atgeirr) [inline src/opm/parser/eclipse/Deck/DeckItem.cpp]: This stands out as problematic. The comment only makes me afraid... I think you should add a comment explaining why it is broken, and what consequences it has. It's ok that the code has limitation, as long as we have some understanding of what they are.

Comment (joakim-hove) [reply src/opm/parser/eclipse/Deck/DeckItem.cpp]: Comment updated.

**Pull request closed without merging**